Framework/API/src/DetectorInfo.cpp

```diff
@@ -93,6 +93,20 @@ Kernel::Quat DetectorInfo::rotation(const size_t index) const {
   return getDetector(index).getRotation();
 }
 
+double DetectorInfo::eFixed(const size_t index) const {
+  const auto &detector = getDetector(index);
+  auto eFixedVector =
+      detector.getNumberParameter(Geometry::ParameterMap::eFixed());
+
+  double eFixed = 0.0;
+
+  if (!eFixedVector.empty()) {
+    eFixed = eFixedVector.front();
+  }
+
```

Comment:
I will go down the enum route with this. The possible cases are:
* Single detector with single eFixed
* Single detector with no eFixed value
* Mulitple detectors each with an eFixed values (return average)
* Multiple detectors with no eFixed values
* Multiple detectors with some eFixed values, some missing eFixed values

There is also the case of all/some detectors not being valid. These should be dealt with in the usual way, with a call to `hasDetectors()` followed by the try/catch used [here](https://github.com/mantidproject/mantid/pull/18034/files#diff-207c349b1b7e9189101af6b77d579965L179).

Not wishing for too many options I would suggest `SUCCESS`, `NO_EFIXED` and `EFIXED_NOT_FOUND_FOR_ALL`.